Replicator/c4ConnectedClient_CAPI.cc

```diff
@@ -57,3 +60,32 @@ void c4client_free(C4ConnectedClient* client) noexcept {
     release(client);
 }
 
+bool c4client_putDoc(C4ConnectedClient* client,
+                        C4Slice docID,
+                        C4Slice collectionID,
+                        C4Slice revID,
+                        C4RevisionFlags revisionFlags,
+                        C4Slice fleeceData,
+                        C4ConnectedClientUpdateDocumentCallback callback,
+                        void * C4NULLABLE context,
+                        C4Error* outError) noexcept {
+    try {
+        bool deletion = (revisionFlags & kRevDeleted) != 0;
+        revidBuffer generatedRev = TreeDocumentFactory::generateDocRevID(fleeceData,
+                                                                         revID,
+                                                                         deletion);
```

Comment:
This should be in the C++ code. The API C functions should just be simple wrappers around the API C++ methods.